### DISCUSSION TITLE
Nutzen einer existierenden InfluxDB

### DISCUSSION THREAD

**Discussion by homematic4me** [solectrus/solectrus#2597] [Q&A] [answered]
Hallo zusammen,
ich finde, Solectrus Dashboard ist genau das was ich schon lange überlegt habe zu machen, an dieser Stelle an riesen Dankeschön an die Entwickler für die geleistete Arbeit!
Nun zu meinem Anliegen: ich habe bereits eine InfluxDB in die ich die Daten von SENEC bzw. EVCC mithilfe von MQTT und Node-red sammle. Jetzt würde ich gerne Solectrus (nur) als Dashboard einsetzen, wäre es möglich die Daten aus der bestehenden Datenbank zu lesen?
Danke und Gruß,
Michael

**Comment by ledermann** [comment]
Danke für die Blumen :)

Eine bestehende Datenbank zu nutzen, wurde in #1132 bereit schon mal diskutiert:

Du kannst in SOLECTRUS zwar BUCKET und MEASUREMENTS deiner InfluxDB konfigurieren, die einzelnen Felder der InfluxDB müssen aber genau in der Form vorhanden sein, wie SOLECTRUS es vorgibt. Also `inverter_power`, `bat_fuel_charge`, `house_power`, `grid_power_plus`, `grid_power_minus` usw. Bei dir heißen die Felder vermutlich anders und möglicherweise bedeuten sie auch etwas anderes.

Da du bereits der zweite bist, der das fragt, will ich das aber nicht einfach abbügeln ;-) Wie genau sieht denn deine InfluxDB-Struktur aus?

**Reply by homematic4me** [reply]
Hallo Georg,
vielen Dank für Deine Antwort!

die InfluxDB-Struktur ist folgende:

- es gibt 2 getrennte Buckets: evcc und senec. Der Grund ist, dass ich zB. aus SENEC noch die einzelnen Stränge überwachen möchte (MPPs) und die Wallbox wiederum nur über EVCC abgreifbar wäre.
- ja, die einzelnen Felder heißen bei mir anders: bat_charge, net_in, net_out, house_consume...

**Reply by ledermann** [reply, marked as answer]
Die Werte für MPP 1-3 werden vom SENEC-Collector bereits jetzt ausgelesen und nach InfluxDB übertragen. SOLECTRUS zeigt es nur noch nicht im Dashboard an, das ist aber [in Planung](#1756).

Nach weiterem Nachdenken über das Thema muss ich aber sagen, dass ich die Möglichkeit der Nutzung einer existierenden InfluxDB in absehbarer Zeit wohl nicht unterstützen werde. Ich sehe da viel Aufwand und potenzielle Probleme, die nur für eine Randgruppe von Nutzern interessant ist. Auch würde es die Weiterentwicklung erschweren, da es eine zusätzliche Daten-Mapping-Schicht geben müsste. 

Vielleicht ist es für dich eine Überlegung wert, SOLECTRUS parallel zu deiner existierenden Installation zu betreiben. Das bedeutet dann zwar doppele Datenhaltung und doppelte Abfrage der Werte vom Speicher, aber möglicherweise ist das zu vernachlässigen.
